src/components/Connection.tsx

```diff
@@ -88,6 +96,10 @@ const Connection: React.FC<ConnectionProps> = ({
   const readerRef = useRef<
     ReadableStreamDefaultReader<Uint8Array> | null | undefined
   >(null); // Ref to store the reader for the serial port
+  const [isPrimary, setIsPrimary] = useState(false);
+
+  // Condition to change the variant
+  const variant = isPrimary ? "primary" : "outline";
```

Comment:
This is of no use, remove this, if error occurs, use ternary operator specifically there.

Reply:
okay